server/chat-commands.js

```diff
@@ -3796,6 +3796,49 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	acceptdraw: 'accepttie',
+	offerdraw: 'offertie',
+	accepttie: 'offertie',
+	offertie(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering ties.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if (cmd === 'accepttie' && !battle.players.some(player => player.wantsTie)) {
+			return this.errorReply("No tie offer available. It might have been automatically withdrawn as the current turn started.");
+		}
+		const playerIds = Object.keys(battle.playerTable);
+		if (!battle.players.some(player => player.wantsTie)) {
+			for (const player of playerIds) {
```

Comment:
Moving the `wantsTie` setting outside of the loop makes sense.

```suggestion
			if (player) player.wantsTie = true;
			for (const otherPlayer of battle.players) {
				if (otherPlayer !== player) {
					otherPlayer.send(Chat.html`|html|${user.name} wants this game to end in a tie; <button class="button" name="send" value="/accepttie">accept tie</button>?`);
				}
			}
```

Reply:
I added this but removed it later. It just adds some form of code complexity because it gets harder to check if it's an offer or an acceptance. I don't feel like (the current code) which is just an ` } else {` is all that bad and probably simpler